cloudera/tests/test_unit_autodiscover.py

```diff
@@ -257,3 +259,835 @@ def test_autodiscover_clusters(
+        'configured host autodiscover with zero hosts',
+        'configured host autodiscover with two different prefix hosts',
+        'configured host autodiscover with ten clusters and limit',
+        'configured host autodiscover with two different prefix hosts and one of them excluded',
+        'configured host autodiscover (with dict) with two different prefix hosts and one of them excluded',
+        'configured host autodiscover with one host when run check two times only one remote call',
+    ],
+    indirect=[],
+)
+def test_autodiscover_hosts(
+    aggregator,
+    dd_run_check,
+    cloudera_check,
+    instance,
+    dd_run_check_count,
+    read_clusters,
+    list_hosts,
+    expected_can_connects,
+    expected_host_healths,
+    expected_metrics,
+):
+    with mock.patch(
+        'datadog_checks.cloudera.client.cm_client.CmClient.get_version',
+        return_value=Version('7.0.0'),
+    ), mock.patch(
+        'datadog_checks.cloudera.client.cm_client.CmClient.read_clusters',
+        return_value=read_clusters,
+    ), mock.patch(
+        'datadog_checks.cloudera.client.cm_client.CmClient.query_time_series',
+        side_effect=query_time_series,
+    ), mock.patch(
+        'datadog_checks.cloudera.client.cm_client.CmClient.list_hosts',
+        return_value=list_hosts,
+    ) as mocked_list_hosts, mock.patch(
+        'datadog_checks.cloudera.client.cm_client.CmClient.read_events',
+        return_value=[],
+    ):
+        check = cloudera_check(instance)
+        for _ in range(dd_run_check_count):
+            dd_run_check(check)
+        for expected_can_connect in expected_can_connects:
+            aggregator.assert_service_check(
+                'cloudera.can_connect',
+                count=expected_can_connect.get('count'),
+                status=expected_can_connect.get('status'),
+                message=expected_can_connect.get('message'),
+                tags=expected_can_connect.get('tags'),
+            )
+        for expected_host_health in expected_host_healths:
+            aggregator.assert_service_check(
+                'cloudera.host.health',
+                count=expected_host_health.get('count'),
+                status=expected_host_health.get('status'),
+                message=expected_host_health.get('message'),
+                tags=expected_host_health.get('tags'),
+            )
+        for expected_metric in expected_metrics:
+            for metric in NATIVE_METRICS['host']:
+                aggregator.assert_metric(
+                    f'cloudera.host.{metric}', count=expected_metric.get('count'), tags=expected_metric.get('tags')
+                )
+            for metric in TIMESERIES_METRICS['host']:
+                aggregator.assert_metric(
+                    f'cloudera.host.{metric}', count=expected_metric.get('count'), tags=expected_metric.get('ts_tags')
+                )
+        mocked_list_hosts.call_count = 1
```

Comment:
Wouldn't the final test case have this `call_count` as 2 since you called the check twice?

Reply:
You are right. I missed the 'assert' there. I will correct it
Good catch!